### PR TITLE
Python3

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

File: plone/rfc822/__init__.py [hunk 1]
```diff
@@ -1,4 +1,13 @@
 # -*- coding: utf-8 -*-
+import sys
+PY3 = sys.version_info[0] == 3
+if PY3:
+    text_type = str
+    binary_type = bytes
+else:
+    text_type = unicode
+    binary_type = str
```
Comment (gforcada) [inline plone/rfc822/__init__.py]: @pbauer isn't [six](https://pypi.python.org/pypi/six) meant for this kind of things?

**Comment by pbauer** [comment]
These are some small changes to allow the tests of plone.supermodel to pass with Python 3. It does not add real Python 3 support to plone.rfc822. 

When @dhavlik and me tried to add support for Python 3 we encoundeted some issues with the email-module in Python 3 and were not able to finish that work. These changes will go into a new branch.

Also: We were asking us what the package is actually used for and found that plone.rfc822 is used in Plone mainly only for two reasons: 
1. The Marker Interface ``IPrimaryField`` is located there. We think there might be better location for that Interface (CMFPlone maybe).
2.  In ``plone.dexterity.filerepresentation`` the core-functionality of the package is used to format dexterity content for input and output via WebDAV.

File: plone/rfc822/fields.rst [hunk 2]
```diff
@@ -86,7 +80,7 @@ This interface is implemented by a the following class:
     ...     _id = 'some.id'
     ...     _dottedName = 'dotted.name'
     ...     _bool = True
-    ...     _int = long(-10)
+    ...     _int = long(-10) if six.PY2 else -10
```
Comment (davilima6) [inline plone/rfc822/fields.rst]: Prefer to check for PY2: /cc @pbauer 

> At the very least, prefer if PY2 to if not PY3. The former stresses that Python 2 is the special case, which is increasingly true going forward. Eventually there’ll be a Python 4, and perhaps even a Python 5, and those future versions will want the “Python 3” behavior.

https://eev.ee/blog/2016/07/31/python-faq-how-do-i-port-to-python-3/#making-some-tough-decisions

**Comment by davilima6** [comment]
WIP: hints are welcome!

**Pull request merged**